common/tqid/task_queue_id.go

```diff
@@ -49,7 +47,7 @@ type (
 	// when starting a worker or a workflow. A task queue family consists of separate TaskQueues for different types of
 	// task (e.g. Workflow, Activity).
 	TaskQueueFamily struct {
-		namespaceId namespace.ID
+		namespaceId string
```

Comment:
This is for getting rid of the dependency. but honestly, not sure I like the namespace.ID and namespace.Name types. they seem unnecessary to me. is mixing up name/id a real problem?

Reply:
I don't really like the types either. Just pointing out what I think you guys already know, which is that Jacob is either working on or going to work on a refactor to make `namespaceId` the only thing used internally, instead of passing both `namespaceName` and `namespaceId` around all the time. Maybe we can have a discussion about the usefulness of typing it on that PR

Reply:
fwiw on this PR my slight preference is to avoid the dependency, and bring it back later if there is confusion